windrose/windrose_main.py

```diff
@@ -0,0 +1,123 @@
+import os
+import pandas as pd
+import numpy as np
+from windrose import WindroseAnalysis
+import time
+from pathlib import Path
+import argparse
+
+
+class WindroseProcessor:
+    def __init__(self, root_directory, directory='windRoseData'):
+        self.root_directory = Path(root_directory)
+        self.directory = Path(directory)
+
+        if not self.directory.exists():
+            self.directory.mkdir(parents=True)
+
+    def process_site_files(self):
+        site_list = [
+            d for d in self.root_directory.iterdir()
+            if d.is_dir()
+        ]
+        # print(site_list)
+
+        for site in site_list:
+            file_list = os.listdir(site)
+            filtered_files = [
+                file for file in file_list
+                if "_BASE_" in file and "_HH_" in file
+            ]
+            # print(site)
+
+            for filtered_file in filtered_files:
+                file_path = site / filtered_file
+                df = pd.read_csv(file_path, skiprows=[0, 1])
+
+                if 'WS' not in df.columns or 'WD' not in df.columns:
+                    print("Both 'WS' and 'WD' or either columns are missing.")
+                else:
+                    self.process_season_data(df, site)
+
+    def process_season_data(self, df, site):
+        df_1 = df[(df['WD'] != -9999) | (df['WS'] != -9999)]
+        speed = df_1['WS']
+        direction = df_1['WD']
+        timestamp = df_1['TIMESTAMP_START']
+        time_series = pd.to_datetime(timestamp, format='%Y%m%d%H%M')
+        hourz = time_series.dt.strftime('%H%M').astype(int)
+        monthz = time_series.dt.month
```

Comment:
what is the reason for appending 'z' to the variable name?

Reply:
I have appended 'z' to the variable name in order to differentiate it from the existing hour and month variable.